components/reddit/actions/list-comments-in-a-post/list-comments-in-a-post.mjs

```diff
@@ -0,0 +1,115 @@
+import reddit from "../../reddit.app.mjs";
+import get from "lodash/get.js";
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "action",
+  key: "reddit-list-comments-in-a-post",
+  version: "0.2.2",
+  name: "List Comments in a Post",
+  description: "List comments for a specific post. [See the docs here](https://www.reddit.com/dev/api/#GET_comments_{article})",
+  props: {
+    reddit,
+    subreddit: {
+      propDefinition: [
+        reddit,
+        "subreddit",
+      ],
+      description: "Search for a subreddit on Reddit with \"Structured Mode\" enabled, or enter a subreddit display name with \"Structured Mode\" disabled (for example, `happycowgifs`).",
+    },
+    subredditPost: {
+      propDefinition: [
+        reddit,
+        "subredditPost",
+        ({ subreddit }) => ({
+          subreddit,
+        }),
+      ],
+    },
+    numberOfParents: {
+      propDefinition: [
+        reddit,
+        "numberOfParents",
+      ],
+    },
+    depth: {
+      propDefinition: [
+        reddit,
+        "depth",
+      ],
+      description: "If set to `1`, include only new comments that are direct children to the subreddit pointed at by **Post**. Furthermore, `depth` determines the maximum depth of children, within the related subreddit comment tree, of new comments to be included.",
+    },
+    includeSubredditDetails: {
+      propDefinition: [
+        reddit,
+        "includeSubredditDetails",
+      ],
+    },
+    sort: {
+      type: "string",
+      label: "Sort",
+      description: "The order of the comments",
+      optional: true,
+      options: [
+        "confidence",
+        "top",
+        "new",
+        "controversial",
+        "old",
+        "random",
+        "qa",
+        "live",
+      ],
+    },
+    after: {
+      propDefinition: [
+        reddit,
+        "after",
+      ],
+    },
+    before: {
+      propDefinition: [
+        reddit,
+        "before",
+      ],
+    },
+    limit: {
+      propDefinition: [
+        reddit,
+        "limit",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const {
+      subreddit,
+      subredditPost,
+      numberOfParents,
+      depth,
+      includeSubredditDetails,
+      sort,
+      after,
+      before,
+      limit,
+    } = this;
+
+    const res = await axios($, this.reddit._getAxiosParams({
+      method: "GET",
+      path: `/r/${get(subreddit, "value", subreddit)}/comments/article`,
```

Comment:
I should have mentioned this before, but ideally every interaction with an app-specific API should go [through the `.app` file](https://github.com/PipedreamHQ/pipedream/blob/9789bd637e61143a5d980b35cff12758fcee5513/docs/docs/components/guidelines/README.md#methods). The `.app` file would encapsulate [knowledge about the API](https://github.com/PipedreamHQ/pipedream/blob/9789bd637e61143a5d980b35cff12758fcee5513/docs/docs/components/guidelines/README.md#good-api-usage) including API URLs and how requests are made (using `axios` vs. a client library). For example, the Twitter app's [Search Mentions](https://github.com/PipedreamHQ/pipedream/blob/master/components/twitter/sources/search-mentions/search-mentions.js) component gets data from the Twitter API through a method in the `.app` file, which can be reused in multiple components.

`spotify.app.mjs` contains [methods](https://github.com/PipedreamHQ/pipedream/blob/4d961d24f78b317186d65100a37fe0acf647ed56/components/reddit/reddit.app.js#L121-L246) used by Reddit sources, which could be reused if updated to support parameters used by the new actions.

Reply:
I would like to maintain this way. We cant use the `$` variable inside the `app` to do some `async options`. If we pass everything to `app`, we may have cases that we need to duplicate the functions and have to `axios`. One from `pipedreamhq` and the regular NPM package. Please let me know if you disagree